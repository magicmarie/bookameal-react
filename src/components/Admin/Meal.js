import React from "react";
import Delete from "./Delete";
import Edit from "./Edit";

// meal card: admin side
const Meal = props => (
  <div className="card mb-2 meal">
    <div className="card-body text-center">
      <div className="card-text"> {props.mealName}</div>
      <div className="card-text">{props.price}</div>
      <hr />
      <div className="row1">
        <button
          to="#"
          data-toggle="modal"
          className="mealcard fa
          fa-plus icon"
          id="add-to-menu"
          aria-hidden="false"
          data-target="#addToMenuModal"
          title="add to menu"
          onClick={() => props.setMealId(props.id)}
        />
        <button
          to="#"
          data-toggle="modal"
          title="edit meal"
          data-target={`#EditModal${props.id}`}
          className="fa fa-edit icon mealcard"
        />
        <button
          to="#"
          id="delete"
          className="fa fa-trash icon mealcard"
          data-toggle="modal"
          title="delete meal"
          data-target={`#deleteModal${props.id}`}
        />
      </div>
      <Delete
        id={props.id}
        meal_name={props.mealName}
        ConfirmDelete={props.ConfirmDelete}
      />
      <Edit
        meal_name={props.mealName}
        id={props.id}
        price={props.price}
        EditMeal={props.EditMeal}
        message={props.message}
      />
    </div>
  </div>
);

export default Meal;
