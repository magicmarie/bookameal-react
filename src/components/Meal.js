import React from "react";
import { Link } from "react-router-dom";
import Delete from "./Delete";
import Edit from "./Edit";

//meal card: admin side
const Meal = props => (
  <div className="card mb-2 meal" style={{ width: "12rem", height: "8rem" }}>
    <div className="card-body text-center">
      <div className="card-text"> {props.mealName}</div>
      <div className="card-text">{props.price}</div>
      <div className="row">
        <Link
          to="#"
          data-toggle="modal"
          className="btn btn-primary btn-sm mealcard"
          data-target="#menuModal"
          onClick={() =>
            props.handleAddMenu(props.id, props.mealName, props.price)
          }
        >
          Add
        </Link>
        <Link
          to="#"
          data-toggle="modal"
          data-target={`#EditModal${props.id}`}
          className="btn btn-success btn-sm mealcard"
        >
          Edit
        </Link>
        <Link
          to="#"
          className="btn btn-danger btn-sm"
          data-toggle="modal"
          data-target={`#deleteModal${props.id}`}
        >
          Delete
        </Link>
      </div>
      <Delete
        id={props.id}
        meal_name={props.mealName}
        ConfirmDelete={props.ConfirmDelete}
      />
      <Edit
        meal_name={props.mealName}
        id={props.id}
        price={props.price}
        EditMeal={props.EditMeal}
        message={props.message}
      />
    </div>
  </div>
);

export default Meal;
