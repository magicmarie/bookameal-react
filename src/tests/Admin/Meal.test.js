import React from "react";
import { shallow } from "enzyme";
import Meal from "../../components/Admin/Meal";

describe("AdminMealCard component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Meal/>);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
