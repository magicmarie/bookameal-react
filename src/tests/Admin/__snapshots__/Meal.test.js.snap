// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminMealCard component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Meal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-body text-center"
>
        <div
                className="card-text"
        >
                 
        </div>
        <div
                className="card-text"
        />
        <hr />
        <div
                className="row1"
        >
                <Link
                        aria-hidden="false"
                        className="mealcard fa fa-plus icon"
                        data-target="#addToMenuModal"
                        data-toggle="modal"
                        onClick={[Function]}
                        replace={false}
                        title="add to menu"
                        to="#"
                />
                <Link
                        className="fa fa-edit icon mealcard"
                        data-target="#EditModalundefined"
                        data-toggle="modal"
                        replace={false}
                        title="edit meal"
                        to="#"
                />
                <Link
                        className="fa fa-trash icon mealcard"
                        data-target="#deleteModalundefined"
                        data-toggle="modal"
                        replace={false}
                        title="delete meal"
                        to="#"
                />
        </div>
        <DeleteMeal
                ConfirmDelete={undefined}
                id={undefined}
                meal_name={undefined}
        />
        <Edit
                EditMeal={undefined}
                id={undefined}
                meal_name={undefined}
                message={undefined}
                price={undefined}
        />
</div>,
      "className": "card mb-2 meal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-text"
>
             
</div>,
          <div
            className="card-text"
/>,
          <hr />,
          <div
            className="row1"
>
            <Link
                        aria-hidden="false"
                        className="mealcard fa fa-plus icon"
                        data-target="#addToMenuModal"
                        data-toggle="modal"
                        onClick={[Function]}
                        replace={false}
                        title="add to menu"
                        to="#"
            />
            <Link
                        className="fa fa-edit icon mealcard"
                        data-target="#EditModalundefined"
                        data-toggle="modal"
                        replace={false}
                        title="edit meal"
                        to="#"
            />
            <Link
                        className="fa fa-trash icon mealcard"
                        data-target="#deleteModalundefined"
                        data-toggle="modal"
                        replace={false}
                        title="delete meal"
                        to="#"
            />
</div>,
          <DeleteMeal
            ConfirmDelete={undefined}
            id={undefined}
            meal_name={undefined}
/>,
          <Edit
            EditMeal={undefined}
            id={undefined}
            meal_name={undefined}
            message={undefined}
            price={undefined}
/>,
        ],
        "className": "card-body text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              undefined,
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card-text",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                aria-hidden="false"
                className="mealcard fa fa-plus icon"
                data-target="#addToMenuModal"
                data-toggle="modal"
                onClick={[Function]}
                replace={false}
                title="add to menu"
                to="#"
/>,
              <Link
                className="fa fa-edit icon mealcard"
                data-target="#EditModalundefined"
                data-toggle="modal"
                replace={false}
                title="edit meal"
                to="#"
/>,
              <Link
                className="fa fa-trash icon mealcard"
                data-target="#deleteModalundefined"
                data-toggle="modal"
                replace={false}
                title="delete meal"
                to="#"
/>,
            ],
            "className": "row1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-hidden": "false",
                "className": "mealcard fa fa-plus icon",
                "data-target": "#addToMenuModal",
                "data-toggle": "modal",
                "onClick": [Function],
                "replace": false,
                "title": "add to menu",
                "to": "#",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "fa fa-edit icon mealcard",
                "data-target": "#EditModalundefined",
                "data-toggle": "modal",
                "replace": false,
                "title": "edit meal",
                "to": "#",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "fa fa-trash icon mealcard",
                "data-target": "#deleteModalundefined",
                "data-toggle": "modal",
                "replace": false,
                "title": "delete meal",
                "to": "#",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "ConfirmDelete": undefined,
            "id": undefined,
            "meal_name": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "EditMeal": undefined,
            "id": undefined,
            "meal_name": undefined,
            "message": undefined,
            "price": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-body text-center"
>
          <div
                    className="card-text"
          >
                     
          </div>
          <div
                    className="card-text"
          />
          <hr />
          <div
                    className="row1"
          >
                    <Link
                              aria-hidden="false"
                              className="mealcard fa fa-plus icon"
                              data-target="#addToMenuModal"
                              data-toggle="modal"
                              onClick={[Function]}
                              replace={false}
                              title="add to menu"
                              to="#"
                    />
                    <Link
                              className="fa fa-edit icon mealcard"
                              data-target="#EditModalundefined"
                              data-toggle="modal"
                              replace={false}
                              title="edit meal"
                              to="#"
                    />
                    <Link
                              className="fa fa-trash icon mealcard"
                              data-target="#deleteModalundefined"
                              data-toggle="modal"
                              replace={false}
                              title="delete meal"
                              to="#"
                    />
          </div>
          <DeleteMeal
                    ConfirmDelete={undefined}
                    id={undefined}
                    meal_name={undefined}
          />
          <Edit
                    EditMeal={undefined}
                    id={undefined}
                    meal_name={undefined}
                    message={undefined}
                    price={undefined}
          />
</div>,
        "className": "card mb-2 meal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-text"
>
               
</div>,
            <div
              className="card-text"
/>,
            <hr />,
            <div
              className="row1"
>
              <Link
                            aria-hidden="false"
                            className="mealcard fa fa-plus icon"
                            data-target="#addToMenuModal"
                            data-toggle="modal"
                            onClick={[Function]}
                            replace={false}
                            title="add to menu"
                            to="#"
              />
              <Link
                            className="fa fa-edit icon mealcard"
                            data-target="#EditModalundefined"
                            data-toggle="modal"
                            replace={false}
                            title="edit meal"
                            to="#"
              />
              <Link
                            className="fa fa-trash icon mealcard"
                            data-target="#deleteModalundefined"
                            data-toggle="modal"
                            replace={false}
                            title="delete meal"
                            to="#"
              />
</div>,
            <DeleteMeal
              ConfirmDelete={undefined}
              id={undefined}
              meal_name={undefined}
/>,
            <Edit
              EditMeal={undefined}
              id={undefined}
              meal_name={undefined}
              message={undefined}
              price={undefined}
/>,
          ],
          "className": "card-body text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                undefined,
              ],
              "className": "card-text",
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card-text",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  aria-hidden="false"
                  className="mealcard fa fa-plus icon"
                  data-target="#addToMenuModal"
                  data-toggle="modal"
                  onClick={[Function]}
                  replace={false}
                  title="add to menu"
                  to="#"
/>,
                <Link
                  className="fa fa-edit icon mealcard"
                  data-target="#EditModalundefined"
                  data-toggle="modal"
                  replace={false}
                  title="edit meal"
                  to="#"
/>,
                <Link
                  className="fa fa-trash icon mealcard"
                  data-target="#deleteModalundefined"
                  data-toggle="modal"
                  replace={false}
                  title="delete meal"
                  to="#"
/>,
              ],
              "className": "row1",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-hidden": "false",
                  "className": "mealcard fa fa-plus icon",
                  "data-target": "#addToMenuModal",
                  "data-toggle": "modal",
                  "onClick": [Function],
                  "replace": false,
                  "title": "add to menu",
                  "to": "#",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "fa fa-edit icon mealcard",
                  "data-target": "#EditModalundefined",
                  "data-toggle": "modal",
                  "replace": false,
                  "title": "edit meal",
                  "to": "#",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "fa fa-trash icon mealcard",
                  "data-target": "#deleteModalundefined",
                  "data-toggle": "modal",
                  "replace": false,
                  "title": "delete meal",
                  "to": "#",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "ConfirmDelete": undefined,
              "id": undefined,
              "meal_name": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "EditMeal": undefined,
              "id": undefined,
              "meal_name": undefined,
              "message": undefined,
              "price": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
